recipes/fast-dds/all/conanfile.py

```diff
@@ -0,0 +1,210 @@
+        if self.settings.os == "Windows":
+            if ("MT" in self.settings.compiler.runtime and self.options.shared):
+                # This combination leads to an fast-dds error when linking
+                # linking dynamic '*.dll' and static MT runtime
+                raise ConanInvalidConfiguration("Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
+
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(self._pkg_share)
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        tools.rename(
+            src=self._pkg_tools,
+            dst=os.path.join(self._pkg_bin, "tools")
+        )
+        tools.remove_files_by_mask(
+            directory=os.path.join(self.package_folder, "lib"),
+            pattern="*.pdb"
+        )
+        tools.remove_files_by_mask(
+            directory=os.path.join(self.package_folder, "bin"),
+            pattern="*.pdb"
+        )
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {"fastrtps": "fastdds::fastrtps"}
+        )
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "fastdds"
+        self.cpp_info.names["cmake_find_multi_package"] = "fastdds"
+        # component fastrtps
+        self.cpp_info.components["fastrtps"].name = "fastrtps"
+        self.cpp_info.components["fastrtps"].libs = tools.collect_libs(self)
+        self.cpp_info.components["fastrtps"].requires = [
+            "fast-cdr::fast-cdr",
+            "asio::asio",
+            "tinyxml2::tinyxml2",
+            "foonathan-memory::foonathan-memory",
+            "boost::boost"
+        ]
+        if self.settings.os in ["Linux", "Macos", "Neutrino"]:
+            self.cpp_info.components["fastrtps"].system_libs.append("pthread")
+        if self.settings.os == "Linux":
+            self.cpp_info.components["fastrtps"].system_libs.extend(["rt", "dl", "atomic"])
+        elif self.settings.os == "Windows":
+            self.cpp_info.components["fastrtps"].system_libs.extend(["iphlpapi","shlwapi"])
+            if self.options.shared:
+                self.cpp_info.components["fastrtps"].defines.append("FASTRTPS_DYN_LINK")
+        if self.options.with_ssl:
+            self.cpp_info.components["fastrtps"].requires.append("openssl::openssl")
+        self.cpp_info.components["fastrtps"].builddirs.append(self._module_subfolder)
+        self.cpp_info.components["fastrtps"].build_modules["cmake_find_package"] = [self._module_file_rel_path]
+        self.cpp_info.components["fastrtps"].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
+        # component fast-discovery
+        self.cpp_info.components["fast-discovery"].name = "fast-discovery"
```

Comment:
it's `fast-discovery-server`

Reply:
Oh Jesus.... Damn.... I would rise an issue and assign it to myself. And in a separate PR resolve the issues.

Reply:
will be resolved in https://github.com/conan-io/conan-center-index/pull/6429